src/lib.rs

```diff
@@ -256,35 +256,30 @@ pub mod contract {
                     return Err(InvariantError::PriceLimitReached);
                 }
 
-                // TODO: refactor
-                let mut tick = Tick::default();
-
-                let update_limiting_tick = limiting_tick.map(|(index, bool)| {
-                    if bool {
-                        tick = self.ticks.get(pool_key, index).unwrap();
-                        (index, Some(&mut tick))
-                    } else {
-                        (index, None)
+                if let Some((tick_index, is_initialized)) = limiting_tick {
```

Comment:
It's hard to determine if it's correct based on github changes, but are you sure if this logic is translated 1:1?